### DISCUSSION TITLE
Reusable Workflows, Jobs outputs

### DISCUSSION THREAD

**Discussion by vad3x** [community/community#26813] [Actions]
<p>I’m trying to implement reusable workflow as described here: <a href="https://docs.github.com/en/actions/learn-github-actions/reusing-workflows" class="inline-onebox" rel="noopener nofollow ugc">Reusing workflows - GitHub Docs</a></p>
<p>on the workflow I have following jobs (simplified):</p>
<pre><code class="lang-auto">jobs:
  get-config:
    name: Get Deployment Config
    runs-on: ubuntu-latest
    outputs:
      config: ${{ steps.get-config.outputs.config }}
    steps:
      - name: Get Config
        uses: action000
        id: get-config

      - run: |
          echo '${{ steps.get-config.outputs.config }}' # returns value

  use-config:
    needs: [get-config]
    name: ccccc
    runs-on: ubuntu-latest
    steps:
      - run: |
          echo '${{ needs.get-config.outputs.config }}' # empty string
</code></pre>
<p>It seems like <code>needs</code> does not work inside of <code>called</code> workflow.</p>
<p>I understand that the docs stands for some limitations:</p>
<blockquote>
<ul>
<li>Outputs generated by a called workflow can’t be accessed by the caller workflow.</li>
</ul>
</blockquote>
<p>Not sure if this is a case given I’m trying to access <code>outputs</code> locally.</p>

**Comment by mforutan** [comment]
<p>Is <code>config</code> a secret? I had this problem trying to output a secret in the first job otherwise it should work as I am using a similar setup in our workflows. In my case there was a line in the raw log complaining about the secret being used in an output.</p>

**Comment by vad3x** [comment]
<p>No, it’s not.</p>
<p>As you can also see that the <code>config</code> output was set successfully on <code>get-config</code> job.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/b/c/bc93626084d50e82fb9259f7aaf05e357b06d596.png" data-download-href="/uploads/short-url/qUdx6053V7oN5CHDRxl8IEDHo8K.png?dl=1" title="Screen Shot 2021-10-04 at 9.38.34 AM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1760687/181100003-7215d417-8e29-49bf-9e47-d5e86a46748b.png" alt="Screen Shot 2021-10-04 at 9.38.34 AM" data-base62-sha1="qUdx6053V7oN5CHDRxl8IEDHo8K" width="690" height="242" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/c/bc93626084d50e82fb9259f7aaf05e357b06d596_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2021-10-04 at 9.38.34 AM</span><span class="informations">718×252 14.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by ChristopherHX** [comment]
<p>I also tried to use job outputs, indeed needs.*.outputs are empty in myjobb. The hardcoded output myin is not a secret so this must be a bug of the github actions beta.</p>
<p><a href="https://github.com/ChristopherHX/newcomposite-sample/runs/3791754085?check_suite_focus=true" rel="noopener nofollow ugc">Here my public workflow run, which prints an empty needs context in myjobb.</a></p>
<p>While running it with <a href="https://github.com/ChristopherHX/runner.server/releases/tag/v3.3.1" rel="noopener nofollow ugc">my own action service implementation locally</a> the same workflow has the expected outputs in the needs context.</p>
<p>I think that mforutan’s experience is about normal workflows, not about reusable workflows. Reusable workflows are documented since last Tuesday.</p>

**Comment by ChristopherHX** [comment, marked as answer]
<p>Fixed as of Today, I can confirm job outputs are now printed to the log</p>
<aside class="onebox githubactions" data-onebox-src="https://github.com/ChristopherHX/newcomposite-sample/actions/runs/1309435631">
  <header class="source">

      <a href="https://github.com/ChristopherHX/newcomposite-sample/actions/runs/1309435631" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Commit">
      <svg width="24" height="24" class="github-icon github-icon-success" viewbox="0 0 24 24" aria-hidden="true"><path d="M1 12C1 5.925 5.925 1 12 1s11 4.925 11 11-4.925 11-11 11S1 18.075 1 12zm16.28-2.72a.75.75 0 00-1.06-1.06l-5.97 5.97-2.47-2.47a.75.75 0 00-1.06 1.06l3 3a.75.75 0 001.06 0l6.5-6.5z"></path></svg>



  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/ChristopherHX/newcomposite-sample/actions/runs/1309435631" target="_blank" rel="noopener nofollow ugc">Update main.yml</a>
    </h4>

      <div class="github-info">
        .github/workflows/main.yml <span class="github-run-number">#28</span>
      </div>
  </div>
</div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p><em>It’s not that github wouldn’t apply changes to the actions-service at any point of time, like between our posts</em></p>

**Comment by vad3x** [comment]
<p>Awesome, works for me too!</p>
